pkg/clusterd/inventory/inventory.go

```diff
@@ -34,21 +34,33 @@ const (
 
 // Discover all the hardware properties for this node.
 // Store the important properties in etcd and more detailed info for the local node in the context.
-func DiscoverHardware(etcdClient etcd.KeysAPI, executor exec.Executor, nodeID string) (*Hardware, error) {
-	devices, err := discoverDevices(executor)
+func DiscoverHardwareAndStore(etcdClient etcd.KeysAPI, executor exec.Executor, nodeID string) (*Hardware, error) {
+	hardware, err := DiscoverHardware(executor)
 	if err != nil {
-		return nil, fmt.Errorf("failed to discover devices. %+v", err)
+		return nil, fmt.Errorf("failed to discover hw. %+v", err)
 	}
 
-	if err = storeDevices(etcdClient, nodeID, devices); err != nil {
+	if err = storeDevices(etcdClient, nodeID, hardware.Disks); err != nil {
 		return nil, fmt.Errorf("failed to store disks in etcd. %+v", err)
 	}
 
-	mem := getSystemMemory()
-	if err = storeMemory(etcdClient, nodeID, mem); err != nil {
+	if err = storeMemory(etcdClient, nodeID, hardware.Memory); err != nil {
 		return nil, fmt.Errorf("failed to store system memory in etcd. %+v", err)
 	}
 
+	return hardware, nil
+}
+
+// Discover all the hardware properties for this node.
+// Store the important properties in etcd and more detailed info for the local node in the context.
+func DiscoverHardware(executor exec.Executor) (*Hardware, error) {
+	devices, err := discoverDevices(executor)
+	if err != nil {
+		return nil, fmt.Errorf("failed to discover devices. %+v", err)
+	}
+
+	mem := getSystemMemory()
+
 	// TODO: Discover network adapters, processors, ...
```

Comment:
is this TODO different than just calling DiscoverHardware like the above DiscoverHardwareAndStore func does?